package/android/src/main/java/com/mrousavy/camera/core/PreviewView.kt

```diff
@@ -45,23 +42,25 @@ class PreviewView(context: Context,
 
   override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
     super.onMeasure(widthMeasureSpec, heightMeasureSpec)
-    val width = MeasureSpec.getSize(widthMeasureSpec)
-    val height = MeasureSpec.getSize(heightMeasureSpec)
-    Log.d(TAG, "onMeasure($width, $height)")
+    val viewWidth = MeasureSpec.getSize(widthMeasureSpec)
+    val viewHeight = MeasureSpec.getSize(heightMeasureSpec)
+    val viewAspectRatio = viewWidth.toDouble() / viewHeight
+
+    if (viewAspectRatio.isNaN()) {
+      return
+    }
+    Log.d(TAG, "onMeasure($viewWidth, $viewHeight) $viewAspectRatio")
+
+    var newWidth: Int = viewWidth
+    var newHeight: Int = viewHeight
 
-    // Performs center-crop transformation of the camera frames
-    val newWidth: Int
-    val newHeight: Int
-    val actualRatio = if (width > height) aspectRatio else 1f / aspectRatio
-    if (width < height * actualRatio) {
-      newHeight = height
-      newWidth = (height * actualRatio).roundToInt()
+    if (viewAspectRatio > aspectRatio) {
+      newWidth = (viewHeight * aspectRatio).roundToInt()
     } else {
-      newWidth = width
-      newHeight = (width / actualRatio).roundToInt()
+      newHeight = (viewWidth / aspectRatio).roundToInt()
     }
 
-    Log.d(TAG, "Measured dimensions set: $newWidth x $newHeight")
+    Log.d(TAG, "Fitted dimensions set: $newWidth x $newHeight")
     setMeasuredDimension(newWidth, newHeight)
   }
```

Comment:
We want to use a center-crop (aspectRatio: 'cover') transform here, not a aspectRatio: 'contain'. The Camera should fill the entire parent View and be horizontally or vertically centered, just like this:

![image](https://github.com/mrousavy/react-native-vision-camera/assets/15199031/046ecea2-e70b-4043-8e1d-51fc26ea10b1)